### ISSUE TITLE
Keras clip_norm could have potential Nan when doing gradient clipping.

### ISSUE THREAD

**Issue by YanLiang1102** [keras-team/keras#14075]
**Potential unsafe behavior in Keras clip_norm**  

https://github.com/keras-team/keras/blob/master/keras/optimizers.py#L39

then_expression = tf.scalar_mul(c / n, g), this line of code when n is inf, this operation is not safe, it will brings in gradient vanishing problem. 
The network I am training when using clipNorm it introduce Inf in the gradients, when using clipValue it works fine, and I figured is the unsafe multiplication in clip_norm might introduce this,
which tensorflow clip_norm does make sure that this operation safe, I think Keras should do that too.

which the tensorflow backend clip_norm implementation address that problem.
```
with ops.name_scope(name, "clip_by_norm", [t, clip_norm]) as name:
    values = ops.convert_to_tensor(
        t.values if isinstance(t, ops.IndexedSlices) else t, name="t")

    # Calculate L2-norm, clip elements by ratio of clip_norm to L2-norm
    l2sum = math_ops.reduce_sum(values * values, axes, keepdims=True)
    pred = l2sum > 0
    # Two-tap tf.where trick to bypass NaN gradients
    l2sum_safe = array_ops.where(pred, l2sum, array_ops.ones_like(l2sum))
    l2norm = array_ops.where(pred, math_ops.sqrt(l2sum_safe), l2sum)
    intermediate = values * clip_norm
    # Assert that the shape is compatible with the initial shape,
    # to prevent unintentional broadcasting.
    _ = values.shape.merge_with(intermediate.shape)
    values_clip = array_ops.identity(
        intermediate / math_ops.maximum(l2norm, clip_norm), name=name)

    if isinstance(t, ops.IndexedSlices):
      return ops.IndexedSlices(values_clip, t.indices, t.dense_shape)

    return values_clip

```
@fchollet

**Comment by jvishnuvardhan** [comment]
@YanLiang1102 This is a stale issue. Sorry for late response. I couldn't locate the code you referred.

Is this still an issue for you? Can you please share a simple standalone code to reproduce the issue? Also, check your code with the recent TF versions (`2.6` and `tf-nightly`) and let us know whether the issue persists with recent TF versions. Thanks!
